src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub/main.tf

```diff
@@ -286,7 +286,9 @@ module "jupyterhub-openid-client" {
   jupyterlab_profiles_mapper = true
   service-accounts-enabled   = true
   service-account-roles = [
-    "view-realm", "view-users", "view-clients"
+    # "manage-clients" is required for creating roles for the client
+    # "manage-users" is required for attaching roles to groups
+    "view-realm", "view-users", "view-clients", "manage-clients", "manage-users"
```

Comment:
Why do we want to add the roles using the service account from the Authentihicator level?

Would it be possible to move the role creation to earlier in the deployment process? For example, in the place where the default roles are being currently created?

Reply:
Like in the similar way the groups are configured:

https://github.com/nebari-dev/nebari/blob/2a1b8772e3a384f5b127f92abff71d3e7ee50ff4/src/_nebari/stages/kubernetes_keycloak_configuration/template/main.tf#L71-L90

My concern is giving too much rights to the service account. It is not necessarily a blocker, I just would like to understand if you went through some thought process when developing it this way.

Reply:
When I started working on it, I added the roles to jupyterhub directly (via authenticator managed roles) and then realised it's not a good idea, as there would be no way to control those at keycloak level. After that the natural progression was to change the code to add roles to keycloak instead. In this process I didn't realize doing it via terraform in the deployment process rather. 

Your concerns are very valid and is actually a blocker. I'll update this to create these at deployment as that's indeed the better way!